### PR TITLE
Test Order

### PR DESCRIPTION
**Pull request by edrisse**

### CODE REVIEW

**Comment by valvijo** [comment]
[9.1.1. Develop component to capture laboratory tests ordered during a clinical visit](https://trello.com/c/EGZLBOPs/33-911-develop-component-to-capture-laboratory-tests-ordered-during-a-clinical-visit)

File: app/clinic/views/patient-laboratory.html [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                </div>
+            </div>
+        </poc-authorize>
+
+
+    </div>
+
+    <div class="col-sm-6">
+
+        <poc-authorize privilege="'Read Drug Prescription'">
```
[the remark this answers is not in the document]
Comment (edrisse) [reply app/clinic/views/patient-laboratory.html]: Privileges fixed. Requires this PR on esaude metadata module https://github.com/esaude/openmrs-module-esaudemetadata/pull/7

File: app/clinic/views/patient-laboratory.html [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                        </div>
+                    </div>
+                </poc-authorize>
+            </form>
+        </div>
+    </div>
+
+    <div class="col-sm-6">
+
+        <poc-authorize privilege="'Read Drug Prescription'">
```
[the remark this answers is not in the document]
Comment (edrisse) [reply app/clinic/views/patient-laboratory.html]: Privileges fixed. Requires this PR on esaude metadata module esaude/openmrs-module-esaudemetadata#7

**Comment by valvijo** [comment]
Because I don't have the privilege to write lab order I can only see this screen:
image
I think that the user should be informed of what is going on. My best option would be not have the Lab Order tab displayed

File: app/clinic/views/patient-laboratory.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <div class="col-sm-6">
         <div>
             <form name="laborderForm" class="form-horizontal" autocomplete="off">
-                <poc-authorize privilege="'Write Drug Prescription'">
+                <poc-authorize privilege="'Write Test Order'">
                     <div class="panel panel-primary panel-marginless-sides">
```
Comment (valvijo) [inline app/clinic/views/patient-laboratory.html]: Where can I find the metadata for these new privileges?

File: app/clinic/controllers/labRequestController.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        testUuids.forEach(function (uuid) {
+          var test = getTestByUuid(uuid);
+          if (test != null) {
+            var alreadyContainsTest = vm.selectedTests.indexOf(test) != -1;
+            if (!alreadyContainsTest) {
+              test.profileName = vm.selectedProfile.name;
+              vm.selectedTests.push(test);
+            }
+          } else {
+            console.error("Teste com uuid " + uuid + " não encontrado");
```
Comment (valvijo) [inline app/clinic/controllers/labRequestController.js]: Tried to add the profile `TESTAGEM VIROLOGIA` and found the following error:
image
However one test was added to the list:
image

Comment (valvijo) [reply app/clinic/controllers/labRequestController.js]: The same for profile `BIOQUIMICA LABORATORIO`
image

Comment (valvijo) [reply app/clinic/controllers/labRequestController.js]: And this after save:
image

Comment (valvijo) [reply app/clinic/controllers/labRequestController.js]: And button `Details` doesn't work:
image

Comment (valvijo) [reply app/clinic/controllers/labRequestController.js]: Some of the profiles are missing a few tests. Namely: `RPR, CD4%, CD4#, Baciloscopia, Triglicerides (TG)`. I think that the errors above are in the source of the problem.
image

**Comment by edrisse** [comment]
Ready for review @valvijo 
Requires https://github.com/esaude/openmrs-module-poc/pull/7

**Comment by valvijo** [comment]
@edrisse are CD4% and CD4# not part of any test profile? I could't find in the three available

**Comment by valvijo** [comment]
Also didn't find `RPR`

**Comment by valvijo** [comment]
Never mind, I didn't find this forth profile `IMMUNOLOGY`

**Comment by valvijo** [comment]
Shall I wait for the updated labels?

**Comment by valvijo** [comment]
Great!

**Pull request merged**